roles/common/tasks/main.yml

```diff
@@ -0,0 +1,7 @@
+# common tasks across multiple roles go here
+- name: Create Network policy if enabled
+  k8s:
+   definition: "{{ lookup('template', 'networkpolicy.yml.j2') | from_yaml }}"
+  vars:
+    trunc_uuid: "{{ cr_state.resources[0].status.suuid }}"
+  when: workload.args.networkpolicy is defined and workload.args.networkpolicy
```

Comment:
As it is written the template would be called every time the reconciliation loop triggered. While it's not really going to break anything we should probably put in a check to see if it exists and then execute if it does not.

something like:
```
block:
  k8s_facts:
      kind: NetworkPolicy
      api_version: networking.k8s.io/v1
      namespace: '{{ operator_namespace }}'
      name: "{{ meta.name }}-networkpolicy-{{ trunc_uuid }}"
    register: net_policy
    
  - name: Create Network policy if enabled
    k8s:
     definition: "{{ lookup('template', 'networkpolicy.yml.j2') | from_yaml }}"
    vars:
      trunc_uuid: "{{ cr_state.resources[0].status.suuid }}"
  when: net_policy | json_query(some_valid_query

 when: workload.args.networkpolicy is defined and workload.args.networkpolicy
```

Thoughts?

Reply:
Yeh it didn't seem like there was a state that i could take advantage of like Running/Created. I''l check again.

Reply:
Done